relayer/relayer.go

```diff
@@ -0,0 +1,170 @@
+package relayer
+
+import (
+	"fmt"
+	"math/big"
+	"net"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/tracker"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+
+	hcf "github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/abi"
+	"github.com/umbracle/ethgo/jsonrpc"
+)
+
+var commitEvent = abi.MustNewEvent(`event NewBundleCommit(uint256 startId, uint256 endId, bytes32 root)`)
+
+type Relayer struct {
+	dataDir           string
+	rpcEndpoint       string
+	stateReceiverAddr ethgo.Address
+	logger            hcf.Logger
+	client            *jsonrpc.Client
+	txRelayer         txrelayer.TxRelayer
+	key               ethgo.Key
+}
+
+func sanitizeRPCEndpoint(rpcEndpoint string) string {
+	if rpcEndpoint == "" || strings.Contains(rpcEndpoint, "0.0.0.0") {
+		_, port, err := net.SplitHostPort(rpcEndpoint)
+		if err == nil {
+			rpcEndpoint = fmt.Sprintf("http://%s:%s", "127.0.0.1", port)
+		} else {
+			rpcEndpoint = "http://127.0.0.1:8545"
+		}
+	}
+
+	return rpcEndpoint
+}
+
+func NewRelayer(
+	dataDir string,
+	rpcEndpoint string,
+	stateReceiverAddr ethgo.Address,
+	logger hcf.Logger,
+	key ethgo.Key,
+) *Relayer {
+	endpoint := sanitizeRPCEndpoint(rpcEndpoint)
+
+	// create the JSON RPC client
+	client, err := jsonrpc.NewClient(endpoint)
+	if err != nil {
+		logger.Error("Failed to create the JSON RPC client", "err", err)
+
+		return nil
+	}
+
+	txRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithClient(client))
+	if err != nil {
+		logger.Error("Failed to create the tx relayer", "err", err)
+	}
+
+	return &Relayer{
+		dataDir:           dataDir,
+		rpcEndpoint:       endpoint,
+		stateReceiverAddr: stateReceiverAddr,
+		logger:            logger,
+		client:            client,
+		txRelayer:         txRelayer,
+		key:               key,
+	}
+}
+
+func (r *Relayer) Start() error {
+	et := tracker.NewEventTracker(
+		path.Join(r.dataDir, "/relayer.db"),
+		r.rpcEndpoint,
+		r.stateReceiverAddr,
+		r,
+		r.logger,
+	)
+
+	return et.Start()
+}
+
+func (r *Relayer) AddLog(log *ethgo.Log) {
+	r.logger.Info("Received a log", "log", log)
+
+	if commitEvent.Match(log) {
+		vals, err := commitEvent.ParseLog(log)
+		if err != nil {
+			panic(err)
+		}
+
+		var startID uint64
+		if sid, ok := vals["startId"].(big.Int); ok {
+			startID = sid.Uint64()
+		}
+
+		var endID uint64
+		if eid, ok := vals["endId"].(big.Int); ok {
+			endID = eid.Uint64()
+		}
+
+		r.logger.Info("Commit", "Block", log.BlockNumber, "StartID", startID, "EndID", endID)
```

Comment:
Nitpick, completely optional, just making log slightly more verbose

```suggestion
		r.logger.Info("NewBundleCommit event captured", "Block", log.BlockNumber, "StartID", startID, "EndID", endID)
```

Reply:
https://github.com/0xPolygon/polygon-edge/pull/1044/commits/9cc552109e99770e8763a7b7aab95828655bc8f2